### PR TITLE
[fix_bug][3.6] stringToBytes, if the type of the input value does not match …

### PR DESCRIPTION
**Pull request by nickAS21**
https://thingsboard-portal.atlassian.net/browse/PROD-2391

##Problem:
### before
 ```
 var msgTest = {
    "hello": "world"
 };
```
- The first: ReflectiveAccessorOptimizer
```
     * org.mvel2.execution.ExecutionHashMap
     *    public String toString() {
     *         String res = super.toString();
     *         return "(id=" + id + ") " + res;
     *     }
```
 **return** _String_ actualStr = "(**id=1) {hello=world}**"
before_1

-  Another: MVELRuntime -> MethodAccessor
     * ! Not have Class[] argParameterTypes = removeExecutionContextParam(parameterTypes); !

before_2

### after
after_02

### CODE REVIEW

File: src/main/java/org/mvel2/ParserConfiguration.java [hunk 1]
```diff
@@ -47,6 +47,7 @@
 public class ParserConfiguration implements Serializable {
 
   protected final Map<String, Object> imports = new ConcurrentHashMap<String, Object>();
+  protected final Map<String, Object> reserved = new ConcurrentHashMap<String, Object>();
```
Comment (ashvayka) [inline src/main/java/org/mvel2/ParserConfiguration.java:50]: This should be either just ```Set<String>```, where the value is TbUtils.getClass().getName().

File: src/main/java/org/mvel2/optimizers/impl/refl/ReflectiveAccessorOptimizer.java [hunk 2]
```diff
@@ -1149,6 +1150,9 @@ else if (ptr instanceof FunctionInstance) {
           cExpr.computeTypeConversionRule();
         }
         if (!cExpr.isConvertableIngressEgress()) {
+          if (pCtx.getParserConfiguration().hasReserved("TbUtils")) {
```
Comment (ashvayka) [inline src/main/java/org/mvel2/optimizers/impl/refl/ReflectiveAccessorOptimizer.java:1153]: still hardcoded value

File: src/main/java/org/mvel2/ParserConfiguration.java [hunk 3]
```diff
@@ -47,6 +47,7 @@
 public class ParserConfiguration implements Serializable {
 
   protected final Map<String, Object> imports = new ConcurrentHashMap<String, Object>();
+  protected final Set<String> reserved = new HashSet<>();
```
Comment (ashvayka) [inline src/main/java/org/mvel2/ParserConfiguration.java:50]: nonConvertableClasses

File: src/main/java/org/mvel2/ParserConfiguration.java [hunk 4]
```diff
@@ -179,6 +180,9 @@ public boolean hasImport(String name) {
         AbstractParser.CLASS_LITERALS.containsKey(name) ||
         checkForDynamicImport(name);
   }
+  public boolean hasReserved(String name) {
```
Comment (ashvayka) [inline src/main/java/org/mvel2/ParserConfiguration.java:183]: name -> clazz

File: src/main/java/org/mvel2/ParserConfiguration.java (hunk 3 above)
Comment (nickAS21) [reply src/main/java/org/mvel2/ParserConfiguration.java:50]: Ok

File: src/main/java/org/mvel2/ParserConfiguration.java (hunk 4 above)
Comment (nickAS21) [reply src/main/java/org/mvel2/ParserConfiguration.java:183]: Ok

**Pull request merged**